packages/mui-material-next/src/ButtonBase/ButtonBase.test.tsx

```diff
@@ -0,0 +1,1106 @@
+        return <a data-testid="customLink" ref={ref} {...other} href={to} />;
+      });
+
+      const { container } = render(
+        // @ts-expect-error missing types in CustomLink
+        <ButtonBase component={CustomLink} to="https://google.com">
+          Hello
+        </ButtonBase>,
+      );
+      const button = container.firstChild;
+      expect(button).not.to.have.attribute('role', 'button');
+    });
+  });
+
+  describe('event callbacks', () => {
+    it('should fire event callbacks', () => {
+      const onClick = spy();
+      const onBlur = spy();
+      const onFocus = spy();
+      const onKeyUp = spy();
+      const onKeyDown = spy();
+      const onMouseDown = spy();
+      const onMouseLeave = spy();
+      const onMouseUp = spy();
+      const onContextMenu = spy();
+      const onDragEnd = spy();
+      const onTouchStart = spy();
+      const onTouchEnd = spy();
+
+      const { getByText } = render(
+        <ButtonBase
+          onClick={onClick}
+          onBlur={onBlur}
+          onFocus={onFocus}
+          onKeyUp={onKeyUp}
+          onKeyDown={onKeyDown}
+          onMouseDown={onMouseDown}
+          onMouseLeave={onMouseLeave}
+          onMouseUp={onMouseUp}
+          onContextMenu={onContextMenu}
+          onDragEnd={onDragEnd}
+          onTouchEnd={onTouchEnd}
+          onTouchStart={onTouchStart}
+        >
+          Hello
+        </ButtonBase>,
+      );
+      const button = getByText('Hello');
+
+      // only run in supported browsers
+      if (typeof Touch !== 'undefined') {
+        const touch = new Touch({ identifier: 0, target: button, clientX: 0, clientY: 0 });
+
+        fireEvent.touchStart(button, { touches: [touch] });
+        expect(onTouchStart.callCount).to.equal(1);
+
+        fireEvent.touchEnd(button, { touches: [touch] });
+        expect(onTouchEnd.callCount).to.equal(1);
+      }
+
+      if (canFireDragEvents) {
+        fireEvent.dragEnd(button);
+        expect(onDragEnd.callCount).to.equal(1);
+      }
+
+      fireEvent.mouseDown(button);
+      expect(onMouseDown.callCount).to.equal(1);
+
+      fireEvent.mouseUp(button);
+      expect(onMouseUp.callCount).to.equal(1);
+
+      fireEvent.contextMenu(button);
+      expect(onContextMenu.callCount).to.equal(1);
+
+      fireEvent.click(button);
+      expect(onClick.callCount).to.equal(1);
+
+      act(() => {
+        button.focus();
+      });
+      expect(onFocus.callCount).to.equal(1);
+
+      fireEvent.keyDown(button);
+      expect(onKeyDown.callCount).to.equal(1);
+
+      fireEvent.keyUp(button);
+      expect(onKeyUp.callCount).to.equal(1);
+
+      act(() => {
+        button.blur();
+      });
+      expect(onBlur.callCount).to.equal(1);
+
+      fireEvent.mouseLeave(button);
+      expect(onMouseLeave.callCount).to.equal(1);
+    });
+  });
+
+  describe('ripple', () => {
+    describe('interactions', () => {
+      it('should not have a focus ripple by default', () => {
```

Comment:
Should we drop this test case if we don't support focus ripple anymore?

Reply:
Yes 🙌🏼